oxidation/crates/parsec_api_protocol/src/error.rs

```diff
@@ -0,0 +1,49 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use parsec_api_types::DateTime;
+use thiserror::Error;
+
+use crate::ApiVersion;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum HandshakeError {
+    #[error("Failed chalenge {0:?}")]
+    FailedChallenge(String),
+    #[error("Handshake bad administration token")]
+    BadAdministrationToken,
+    #[error("Handshake bad identity")]
+    BadIdentity,
+    #[error("Handshake organization expired")]
+    OrganizationExpired,
+    #[error("Handshake rvk mismatch")]
+    RVKMismatch,
+    #[error("Handshake revoked device")]
+    RevokedDevice,
+    #[error("Handshake out of ballpark {0:x?}")]
+    OutOfBallpark(ChallengeDataReport),
+    #[error("No overlap between client API versions {client_versions:?} and backend API versions {backend_versions:?}")]
+    APIVersion {
+        client_versions: Vec<ApiVersion>,
+        backend_versions: Vec<ApiVersion>,
+    },
+    #[error("{0}")]
+    InvalidMessage(String),
+    #[error("`verify_key` param must be provided for authenticated handshake")]
+    Authenticated,
```

Comment:
`Authenticated` is too broad name, `MissingVerifyKey` seems better considering the error message.

And why not instead considering this is a generic error and use ```InvalidMessage("`verify_key` param must be provided for authenticated handshake")``` ?